src/components/App.js

```diff
@@ -5,11 +5,37 @@ import Main from './Main';
 import Footer from './Footer';
 import Modal from './Modal';
 
+import PageNotFound from './PageNotFound';
+import {
+    BrowserRouter as Router,
+    Switch,
+    Route
+} from "react-router-dom";
+import MovieDetails from './movie/MovieDetails';
+
 const App = () => {
     return (
         <React.Fragment>
-            <Header/>
-            <Main/>
+            <Router>
+                <Switch>
+                    <Route exact path="/">
+                        <Header/>
+                        <Main/>
+                    </Route>
+                    <Route path="/search/:SearchQuery">
+                        <Header/>
+                        <Main/>
+                    </Route>
```

Comment:
This part seems to break DRY principle (don't repeat yourself).

We certainly understand that `Header` and `Footer` components should be on every page so we can create component called `Layout` for example, and put it there like this:

```
const Layout = ({ children }) => (
  <div>
    <Header />
    <main>
      {children}
    </main>
    <Footer />
  </div>
);
``` 

and use it like this:

```
    <Router>
    <Layout>
        <Switch>
            ALL YOUR ROUTES GO HERE
        </Switch>
    </Layout>
    </Router>
```

Next, we need to have 3 pages here:
- **NotFound page** - that will be shown for every non existent page

```
<Route path="*">
      <PageNotFound/>
  </Route>
```

or another way to pass component is `<Route path="*" component={NotFound} />`

- **Movie details page `film/:id`:**
`<Route path="/film/:id" component={MovieDetailsPage} />` and this `MovieDetailsPage` can be a composition of two components: `MovieDetails` and `MovieList`

```
const MovieDetailsPage = () => (
  <>
    <MovieDetails />
    <MovieList />
  </>
);
```

- **same thing is applicable for our third page (`/search/:query`)**


With the changes above we can guarantee that we will have clear and readable routes, we do not repeat code, and with any further visible changes within components - file with routes won't be changed

Reply:
Fixed, thank you.